src/rascal/representations/calculator_spherical_invariants.hh

```diff
@@ -522,37 +738,35 @@ namespace rascal {
           if (spair_type[0] > el2.first[0]) {
             continue;
           }
+
           spair_type[1] = el2.first[0];
           auto & coef2{el2.second};
-          auto && soap_vector_by_pair{soap_vector[spair_type]};
-
-          size_t n1n2{0};
-          size_t pos{0}, size{0};
-          for (size_t n1{0}; n1 < this->max_radial; ++n1) {
-            for (size_t n2{0}; n2 < this->max_radial; ++n2) {
-              soap_vector_by_pair(n1n2, 0) = coef1(n1, 0) * coef2(n2, 0);
-              pos = 1;
-              for (size_t l{1}; l < this->max_angular + 1; ++l) {
-                size = 2 * l + 1;
-                // do the reduction over m (with vectorization)
-                soap_vector_by_pair(n1n2, l) =
-                    (coef1.block(n1, pos, 1, size).array() *
-                     coef2.block(n2, pos, 1, size).array())
-                        .sum();
-                pos += size;
-              }
-              ++n1n2;
+          auto && soap_vector_by_pair{soap_vector[this->key_map[spair_type]]};
+          const auto & coef_ids{coeff_indices_map[spair_type]};
+          for (const auto & coef_idx : coef_ids) {
+            // multiply with the constant 1 / \sqrt(2l+1)
```

Comment:
> as the actual multiplication with the l-factor doesn't happen until several lines down.

How so ? it is done in the next line, just clang-format did put a few line break in the mix for some reasons.

Reply:
It was enough line breaks to put the comment quite far from the line it's describing... but what's there now is fine, because it describes all of what's happening in the next few lines.